src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -109,22 +109,30 @@ class CoursierSubsystem(TemplatedExternalTool):
     name = "coursier"
     help = "A dependency resolver for the Maven ecosystem. (https://get-coursier.io/)"
 
-    default_version = "v2.1.0-M5-18-gfebf9838c"
+    default_version = "v2.1.6"
+
+    _X86_64_URL_TEMPLATE = (
+        "https://github.com/coursier/coursier/releases/download/{version}/cs-{platform}.gz"
+    )
+    _ARM64_URL_TEMPLATE = (
+        "https://github.com/VirtusLab/coursier-m1/releases/download/{version}/cs-{platform}.gz"
+    )
+
+    default_url_template = _X86_64_URL_TEMPLATE
```

Comment:
Another option that may be more honest to the user is to have this default_url_template present as something like:
- `"<depends on platform>"`
- `"https://github.com/{coursier/coursier | VirtusLab/coursier-m1}/releases/download/{version}/cs-{platform}.gz"`
- `"https://github.com/coursier/coursier/releases/download/{version}/cs-{platform}.gz for x86, https://github.com/VirtusLab/coursier-m1/releases/download/{version}/cs-{platform}.gz for arm"`

It will show up when you look at help for the url_template option. Unfortunately, it will confuse the user into thinking they could use some kind of syntax like that to submit the option themselves.

Alternatively, could special-case the whole thing and add an explicit new Option: `url_template_arm64` just for the `CoursierSubsystem`. That would be the clearest.

Thoughts?